Controller/ProductFaq/Create.php

```diff
@@ -118,22 +146,30 @@ public function __construct(
      */
     public function execute()
     {
-        $this->session->authenticate();
+        if (!$this->config->getProductFaqActive()) {
+            return $this->forwardFactory->create()->forward('noroute');
+        }
+
+        /** @var Redirect $resultRedirect */
+        $resultRedirect = $this->resultFactory->create(ResultFactory::TYPE_REDIRECT);
+
+        if (!$this->customerSession->authenticate()) {
+            $resultRedirect->setPath($this->customerUrl->getLoginUrl());
+            return $resultRedirect;
+        }
 
         $data = $this->request->getParams();
-        $storeId = $this->storeManager->getStore()->getId();
 
         try {
-            $product = $this->productRepository->getById($data['product_id'], false, $storeId);
+            $product = $this->productRepository->getById($data['product_id']);
         } catch (NoSuchEntityException $e) {
             $this->messageManager->addErrorMessage(__('Product was not found.'));
             return $this->forwardFactory->create()->forward('noroute');
         }
 
-        $resultRedirect = $this->resultFactory->create(ResultFactory::TYPE_REDIRECT);
         $resultRedirect->setPath($product->getProductUrl() . self::PRODUCT_FAQ_TAB_URL_SUFFIX);
 
-        if (strlen($data['question']) < self::QUESTION_MIN_LENGTH) {
+        if (mb_strlen($data['question']) < self::QUESTION_MIN_LENGTH || empty($data['question'])) {
```

Comment:
možda ovako, ako trimaš prije 
```suggestion
        $question = trim($this->request->getParam('question', ''));
        if (strlen(question) < self::QUESTION_MIN_LENGTH) {
```